src/complex.cr

```diff
@@ -36,6 +39,48 @@ struct Complex
     false
   end
 
+  # Returns `self`.
+  def to_c
+    self
+  end
+
+  # Returns the value as a `Float64` if possible (the imaginary part should be exactly zero),
+  # raises otherwise.
+  def to_f64
+    unless @imag.zero?
+      raise Exception.new "Complex number with non-zero imaginary part can't be converted to real number"
```

Comment:
But what's the point of these methods then? Why not just provide an `assert_real!` and an `assert_imaginary!` method and let people use `c.real`? What's the usecase of a number conversion method that raises all the time?

Reply:
> But what's the point of these methods then? Why not just provide an assert_real! and an assert_imaginary! method and let people use c.real? What's the usecase of a number conversion method that raises all the time?

Somehow that's what Ruby does, so I guess all of your questions should be directly answered by digging into the Ruby land, their decision making and API designing process.

> I seriously think this is a badly thought out PR trying to shoehorn a Complex into being something it's not: a single-dimensional dumber.

And what's the point of of your criticism? I don't see any constructive arguments, only your personal opinions which you hold dearly to the point of putting yourself on airs.

> If it doesn't fit in the interface for Number, it's not a Number regardless of if it's a number. Yes that points to our number hierarchy being inconsistent with pure mathematics but i'm not sure how important that is in the real world, or if it's worth adding many new interfaces for it like I think Rust has done.

AFAIK the only thing from `Number` API it doesn't fit is `<=>` operator which IMO shouldn't be included there from default.